src/custom/lib/hooks/routing/clientSideSmartOrderRouter/clientSideSmartOrderRouterMod.ts

```diff
@@ -10,3 +10,7 @@ export function isSupportedChainId(chainId: ChainId | undefined): boolean {
   if (chainId === undefined) return false
   return toSupportedChainId(chainId) !== undefined
 }
+
+export function isSupportedChainIdType(chainId: number): chainId is SupportedChainId {
```

Comment:
Nitpick: Can we just change the return type of `isSupportedChainId`? `boolean` => `chainId is SupportedChainId`

Reply:
I tried that, but it didn't work. I took a bad assumption on why it didn't

Now i tried again, and see it works if I also change the type of the param to `chainId: number | undefined`

Reply:
I'm addressing in a new PR on top of the DELETION of code to avoid messing it up

8b02fee64a18ed0295896fb0854e96571dc4bc34